chia/simulator/full_node_simulator.py

```diff
@@ -144,3 +173,219 @@ async def reorg_from_index_to_new_index(self, request: ReorgProtocol):
+            height = uint32(block.height)
+            rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
+
+        while True:
+            peak_height = self.full_node.blockchain.get_peak_height()
+            if peak_height is None:
+                raise RuntimeError("Peak height still None after processing at least one block")
+
+            if peak_height >= height:
+                break
+
+            await asyncio.sleep(0.050)
+
+        return rewards
+
+    async def farm_blocks(self, count: int, wallet: Wallet) -> int:
+        """Farm the requested number of blocks to the passed wallet. This will
+        process additional blocks as needed to process the reward transactions
+        and also wait for the rewards to be present in the wallet.
+
+        Arguments:
+            count: The number of blocks to farm.
+            wallet: The wallet to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos farmed to the requested address.
+        """
+        if count == 0:
+            return 0
+
+        rewards = await self.process_blocks(count=count, farm_to=await wallet.get_new_puzzlehash())
+        await self.process_blocks(count=1)
+
+        peak_height = self.full_node.blockchain.get_peak_height()
+        if peak_height is None:
+            raise RuntimeError("Peak height still None after processing at least one block")
+
+        coin_records = await self.full_node.coin_store.get_coins_added_at_height(height=peak_height)
+        block_reward_coins = {record.coin for record in coin_records}
+
+        await wait_for_coins_in_wallet(coins=block_reward_coins, wallet=wallet)
+
+        return rewards
+
+    async def farm_rewards(self, amount: int, wallet: Wallet) -> int:
+        """Farm at least the requested amount of mojos to the passed wallet. Extra
+        mojos will be received based on the block rewards at the present block height.
+        The rewards will be present in the wall before returning.
+
+        Arguments:
+            amount: The minimum number of mojos to farm.
+            wallet: The wallet to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos farmed to the requested wallet.
+        """
+        rewards = 0
+
+        if amount == 0:
+            return rewards
+
+        height_before = self.full_node.blockchain.get_peak_height()
+        if height_before is None:
+            height_before = uint32(0)
+
+        for count in itertools.count(1):
+            height = uint32(height_before + count)
+            rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
+
+            if rewards >= amount:
+                await self.farm_blocks(count=count, wallet=wallet)
+                return rewards
+
+        raise Exception("internal error")
+
+    async def wait_transaction_records_entered_mempool(self, records: Collection[TransactionRecord]) -> None:
```

Comment:
Maybe add a timeout to prevent 60min long CI runs

Reply:
Consideration of timeouts is indeed on the draft reason list in the OP.  https://github.com/Chia-Network/chia-blockchain/pull/11535#issue-1236288585